Makefile

```diff
@@ -208,3 +208,14 @@ python-build:
 python-test:
 	$(MAKE) python-build extras="[tests]"
 	DEBUG=1 ${PYTHON_EXECUTABLE} -m unittest -v
+
+clang-debug:				#: Build with debugging symbols using Clang
```

Comment:
Is there a reason we are having compiler specific targets here ? i think we should avoid this and move this logic to the cmakefile. This can possibly lead to changes in docker scripts, documentation , and possible user confusion and difference in behavior if say you just used a different target.

Reply:
This target wraps the normal release and debug targets. I thought it was the easiest way for a user to decide what they want to do at this point.

We still would need to make changes in the Makefile to have an environment variable or something that decides what to use flow into the CMakeLists.txt.

One of the other reasons was that eventually we'll just default to clang and we don't need a compiler selection process anymore. It would be hard coded in the CMakeLists.txt and these targets removed.

Reply:
Cmake will use the CC, CXX as env variables to determine the compiler ; This is pretty standard and cmake has long supported it. One can just set this env variable and have cmake infer the rest. I think thats proabaly a better way to go : https://gitlab.kitware.com/cmake/community/-/wikis/FAQ#how-do-i-use-a-different-compiler

Reply:
Right, and it was used this way in the setup scripts. A user can manually set the CC/CXX to switch to clang.